_pytest/logging.py

```diff
@@ -371,18 +376,20 @@ def _runtest_for(self, item, when):
                            formatter=self.formatter, level=self.log_level) as log_handler:
             if self.log_cli_handler:
                 self.log_cli_handler.set_when(when)
-            if not hasattr(item, 'catch_log_handlers'):
+            if item and not hasattr(item, 'catch_log_handlers'):
                 item.catch_log_handlers = {}
-            item.catch_log_handlers[when] = log_handler
-            item.catch_log_handler = log_handler
+            if item:
+                item.catch_log_handlers[when] = log_handler
+                item.catch_log_handler = log_handler
             try:
                 yield  # run test
             finally:
-                del item.catch_log_handler
+                if item:
+                    del item.catch_log_handler
                 if when == 'teardown':
```

Comment:
This should be indented (under `if item:`) right?

Reply:
Not necessarily because item does exist on teardown. But it can be moved.

Reply:
Oh you are right, well up to you!